### DISCUSSION TITLE
Connecting to Firebase realtime database

### DISCUSSION THREAD

**Discussion by loucimj** [skiptools/skip#112] [Q&A] [answered]
Hello! I've been reviewing the skip database wrapper https://github.com/skiptools/skip-firebase/blob/main/Sources/SkipFirebaseDatabase/SkipFirebaseDatabase.swift and it does not implement the reference API to obtain and send data to the realtime database.

Is this planned to be rolled out? Is anybody with the same problem?

Thanks!
Javier

**Comment by marcprux** [comment, marked as answer]
Currently only the Firestore component of the skip-firebase package is implemented. The implementation can be seen at: https://github.com/skiptools/skip-firebase/blob/main/Sources/SkipFirebaseFirestore/SkipFirebaseFirestore.swift

As you can see, it is a pretty straightforward 1-1 mapping between the Swift and Firebase APIs, thanks to the homogeneous design of the official Firebase Swift (https://firebase.google.com/docs/reference/swift/firebasedatabase/api/reference/Classes/Database) and Kotlin (https://firebase.google.com/docs/reference/android/com/google/firebase/database/FirebaseDatabase) frameworks. If you'd like to undertake implementing some of it yourself, we'd certainly appreciate any PRs you can provide.

Otherwise, we can add it to our priority list. Are there any specific aspects of the library that are more important to you than others?
